In [78]:
import numpy as np
import pandas as pd
from scipy.stats import t

### __Question 1__

A factory claims avg. wieight of cereal box is 500 grams. A sample of 64 boxes is taken at random with a mean of 490 grams and population standard deviation is 15 grams (sigma) with significance level at 5%. Determine the validity of this claim using hypothesis testing.

#### __Solution:__

- It is a two-tailed test as it is not mentioned if it is a one-tailed test.
- It is a one-sample test.
- It qualifies for Z-test as sample size > 30 and population standard deviation is known.
- Since, it is a two-tailed test, Z-score will be (α / 2) = (0.05 / 2) = 0.025.
- So, Z value will be from __-1.96__ to __1.96__. ($Z_{critical}$)

In [24]:
x_bar = 490
mu = 500
sigma = 15
n = 64

Z = (x_bar - mu) / (sigma/np.sqrt(n))
Z

-5.333333333333333

##### __Observation:__

- Here, __Z statistic__ is coming as -5.33 which lies outside the acceptance region (i.e less than) between `-2.576` to `2.576`.
- Hence, we can reject the null hypothesis.
- So, the claim made by the company stating that the average weight of the box being 500 grams is incorrect.

### __Question 2__

A company claims the average battery life of their batteries is 300  hours. A sample of 100 batteries is taken at random and they show average of 295 hours and population standard deviation of 10 hours at significance level of 1%. Determine the validity of this claim using hypothesis testing.

#### __Solution:__

- It is a two-tailed test as it is not mentioned if it is a one-tailed test.
- It is a one-sample test.
- It qualifies for Z-test as sample size > 30 and population standard deviation is known.
- Since, it is a two-tailed test, Z-score will be (α / 2) = (0.01 / 2) = 0.005.
- So, Z value will be __-2.57__ to __2.57__. ($Z_{critical}$)

* $H_0$ = 300
* $H_A$ $\neq$ 300

In [29]:
x_bar = 295
mu = 300
sigma = 10
n = 100

Z = (x_bar - mu) / (sigma/np.sqrt(n))
Z

-5.0

##### __Observation:__

- Here, __Z statistic__ is coming as -5 which lies outside the acceptance region (i.e. less than) between `-2.57` to `2.57`.
- Hence, we can reject the null hypothesis.
- So, the claim made by the company stating that the average battery life of their batteries being 300 hours is incorrect.

### __Question 3__

In a company, there is a Group A which has 40 employees and average productivity of 120 units and standard deviation of 15. Group B has 50 employees with average productivity of 125 units and standard deviation of 20. Is there a significant difference between their productivities at a significance level of 5%.

#### __Solution:__

- It is a two-tailed test as it is not mentioned if it is a one-tailed test.
- It is a two-sample test.
- It qualifies for Z-test as sample size > 30 and population standard deviation is known.
- Since, it is a two-tailed test, Z-score will be (α / 2) = (0.05 / 2) = 0.025.
- So, Z value will be __-1.96__ to __1.96__. ($Z_{critical}$)

In [34]:
x_bar1 = 120
sigma_1 = 15
n_1 = 40

x_bar2 = 125
sigma_2 = 20
n_2 = 50

Z = (x_bar1 - x_bar2) / np.sqrt(np.square(sigma_1)/n_1 + np.square(sigma_2)/n_2)
Z

-1.3545709229571927

In [36]:
-1.35 > -1.96

True

##### __Observation:__

- Here, __Z statistic__ is coming as -1.36 which lies inside the acceptance region or confidence interval between `-1.96` to `1.96`.
- Hence, we fail to reject the null hypothesis, i.e. accept the null hypothesis.
- So, there is no significant difference between the two groups' productivity, i.e. they are working at the same productivity level.

### __Question 4__

Medicine A reduces b.p  by 10 mmHg for a sample size of 60 and standard deviation at 2. Medicine B reduces b.p by 12 mmHg for a sample size of 70 and standard deviation at 2.5. Significance level is 5%.

#### __Solution:__

- It is a two-tailed test as it is not mentioned if it is a one-tailed test.
- It is a two-sample test.
- It qualifies for Z-test as sample size > 30 and population standard deviation is known.
- Since, it is a two-tailed test, Z-score will be (α / 2) = (0.05 / 2) = 0.025.
- So, Z value will be __-1.96__ to __1.96__. ($Z_{critical}$)

In [41]:
x_bar1 = 10
sigma_1 = 2
n_1 = 60

x_bar2 = 12
sigma_2 = 2.5
n_2 = 70

Z = (x_bar1 - x_bar2) / np.sqrt(np.square(sigma_1)/n_1 + np.square(sigma_2)/n_2)
Z

-5.064469859899658

##### __Observation:__

- Here, __Z statistic__ is coming as -5.06 which lies inside the acceptance region or confidence interval between `-1.96` to `1.96`.
- Hence, we reject the null hypothesis.
- So, there is no significant difference between the two medicines' effectiveness, i.e. they are same effect in reducing the blood pressure of the patients.

### __Question 5__

An organisation claims that a new battery lasts longer than 50 hours on  average. A researcher collects 30 samples and the sample average hours is 52. Population std. deviation is 4 hours. Significance level is 5%. Perform hypothesis testing to verify the claim. Perform a Right-tailed test.

#### __Solution:__

- It is a right-tailed test.
- It is a one-sample test.
- It qualifies for Z-test as sample size > 30 and population standard deviation is known.

In [46]:
x_bar = 52
mu = 50
sigma = 4
n = 30

Z = (x_bar - mu) / (sigma/np.sqrt(n))
Z

2.7386127875258306

##### __Observation:__

- Here, Z value for 2.73 is 0.032.
- P-value will be 0.0032 as it is right-tailed.
- Alpha is 0.05.
- Here, we are considering area under the curve and 0.0032 < 0.05, hence we reject the null hypothesis.
- So, the claim made by the company about 50 hours average battery life is incorrect. 

### __Question 6__

A teacher claims that average score of students in class is 80. 10 students were picked up at random and average scores were $[75,78,82,79,81,77,84,76,80,83]$. At significance level of 5%. Peform hypothesis testing that the claim is correct.

- Null Hypothesis: mu = 80
- Alternate Hypothesis: mu $\neq$ 80

#### __Solution:__

- It is a two-tailed and 1-sample test.
- It qualifies for T-test as sample size is < 30 and population std. deviation is not known.
- Since, it is a two-tailed test, degrees of freedom (df) will be 9, alpha = 0.05, T-critical will be 2.262.
- So, T value will be __-2.262__ to __2.262__. ($T_{critical}$)

In [134]:
scores = np.array([75,78,82,79,81,77,84,76,80,83])
x_bar = np.mean(scores)
mu = 80
n = 10
s = np.std(scores)
alpha = 0.05

T = (x_bar - mu) / (s/np.sqrt(n))
print("t: ", T)

# Since, it is a two-tailed test, we need to multiply the p_value twice
p_value = t.cdf(T, df = n - 1) * 2
print("p_value: ", p_value)

if p_value < alpha:
    print("Reject the null hypothesis. The sample mean is significantly different from the population mean.")
else:
    print("Fail to reject the null hypothesis. No significant difference between the sample and population mean.")

t:  -0.5504818825631803
p_value:  0.5953858958226808
Fail to reject the null hypothesis. No significant difference between the sample and population mean.


##### __Observation:__

- T = -0.55, alpha = 0.05
- Since, T is falling inside the T critical range, we will accept the null hypothesis.
- Since p_value > alpha, it means area under the curve is greater than the threshold, hence we fail to reject the null hypothesis.
- So, the claim made by the teacher regarding average scores being 80 is correct.

### __Question 7__

A researcher claims that time to solve a puzzle is less than 15 minutes. 10 participants were picked up at random and sample mean to solve puzzle is 14 mins and standard deviation is 2 mins. Perform hypothesis testing to verify the claim. Significance level is 5%.

- Null Hypothesis: mu $\geq$ 15 mins
- Alternate Hypothesis: mu < 15 mins

#### __Solution__

- It is a 1-sample test.
- It is a left-tailed test as we are checking the value less than the mean as part of alternative hypothesis.

In [132]:
x_bar = 14
mu = 15
n = 10
s = 2
alpha = 0.05

T = (x_bar - mu) / (s/np.sqrt(n))
print("t: ", T)

p_value = t.cdf(T, df = n - 1)
print("p_value: ", p_value)

if p_value < alpha:
    print("Reject the null hypothesis. The sample mean is significantly different from the population mean.")
else:
    print("Fail to reject the null hypothesis. No significant difference between the sample and population mean.")

t:  -1.5811388300841895
p_value:  0.07415235368327971
Fail to reject the null hypothesis. No significant difference between the sample and population mean.


##### __Observation:__

- p_value is 0.074, alpha is 0.05
- Since p_value > alpha, it means area under the curve is greater than the threshold, hence we fail to reject the null hypothesis.
- Hence, we accept the null hypothesis.
- So, the claim made by the researcher about mean puzzle time being 15 mins is correct.

### __Question 8__

A company claims that a worker on an average earns 50 dollars/hour. 8 workers were sampled at random and their salaries were $[48,51,49,50,47,52,49,48]$. Significance level is 1%. Test the claim using hypothesis testing.

- Null hypothesis: mu = 50
- Alternative hypothesis: mu $\neq$ 50

#### __Solution:__

- It is a 1-sample and two-tailed test.
- Using T-test.
- Alpha = 0.01

In [140]:
salaries = np.array([48,51,49,50,47,52,49,48])
x_bar = np.mean(salaries)
mu = 50
s = np.std(salaries)
n = 8
alpha = 0.01

T = (x_bar - mu) / (s/np.sqrt(n))
print("t: ", T)

# Since, it is a two-tailed test, we need to multiply the p_value twice
p_value = 2 * t.cdf(T, df = n - 1)
print("p_value: ", p_value)

if p_value < alpha:
    print("Reject the null hypothesis. The sample mean is significantly different from the population mean.")
else:
    print("Fail to reject the null hypothesis. No significant difference between the sample and population mean.")

t:  -1.358732440973515
p_value:  0.21638711240256153
Fail to reject the null hypothesis. No significant difference between the sample and population mean.


##### __Observation:__

- p_value is 0.211, alpha is 0.01
- Since p_value > alpha, it means area under the curve is greater than the threshold, hence we fail to reject the null hypothesis.
- Hence, we accept the null hypothesis.
- So, the claim made by the company regarding average worker salary as 50 dollars/hour is correct.